src/test/scala/mesosphere/marathon/api/v2/AppsResourceTest.scala

```diff
@@ -252,6 +264,191 @@ class AppsResourceTest extends MarathonSpec with MarathonActorSupport with Match
     response.getEntity.toString should include("must not be empty")
   }
 
+  test("Creating an app with an external volume for an illegal provider should fail") {
+    Given("An app invalid volume (illegal volume provider)")
+    val response = createAppWithVolumes("MESOS",
+      """
+        |    "volumes": [{
+        |      "containerPath": "/var",
+        |      "external": {
+        |        "size": 10,
+        |        "name": "foo",
+        |        "provider": "acme"
+        |      },
+        |      "mode": "RW"
+        |    }]
+      """.stripMargin
+    )
+
+    Then("The return code indicates that the hostPath of volumes[0] is missing") // although the wrong field should fail
+    response.getStatus should be(422)
+    response.getEntity.toString should include("/container/volumes(0)/external/providerName")
+    response.getEntity.toString should include("is unknown provider")
+  }
+
+  test("Creating an app with an external volume with no name provider name specified should FAIL provider validation") {
+    Given("An app with an unnamed volume provider")
+    val e = intercept[JsResultException]{
```

Comment:
Well, this is good enough for now.

Basically, we have two ways to indicate errors: One is throwing an exception and one is returning an error response. Ultimately, in a running Marathon instance, all exceptions are also mapped to a response (see `MarathonExceptionMapper`). This is done in Chaos-Framework-Code.

If you wanted to emulate that behavior, you could do something like (probably factor that out into a `MarathonExceptionMapperTestHelper` or similar):

```scala
    val response = Try {
      createAppWithVolumes("MESOS",
        """
          |    "volumes": [{
          |      "containerPath": "/var",
          |      "external": {
          |        "size": 10
          |      },
          |      "mode": "RW"
          |    }]
        """.stripMargin
      )
    }.recover { case NonFatal(e: Exception) => (new MarathonExceptionMapper).toResponse(e) }.get
```

It is part of your testing philosophy if you want thus to extend the scope of your testing to include other components or not.